### PR TITLE
update js deps

### PR DESCRIPTION
**Pull request by happysalada**
### Changes

Updates several js dependencies

There are more updates available (react, eslint) but I don't know enough about those.

One notable change is that `npm run delpoy` will yield a new warning for app.css
before this PR on my local it yields 5.8 Mb
after the PR it goes up to 6.07 Mb
don't know if that is an issue just mentioning it.

There might be reasons for why some  dependencies have been kept back, feel free to let me know.

### Tests
- [ ] Automated tests have been added
- [X] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [X] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [X] This change does not need a documentation update

### CODE REVIEW

File: assets/package.json [hunk 1]
```diff
@@ -6,38 +6,38 @@
...
     "phoenix": "file:../deps/phoenix",
     "phoenix_html": "file:../deps/phoenix_html",
-    "postcss": "^7.0.35",
-    "postcss-loader": "^4.0.4",
+    "postcss": "^8.3.0",
+    "postcss-loader": "^5.3.0",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-flatpickr": "^3.10.5",
+    "react-flatpickr": "3.10.5",
```
Comment (happysalada) [inline assets/package.json:32]: leaving the caret, it will try to install 3.10.7 which requires react 17

File: assets/package.json [hunk 2]
```diff
@@ -6,38 +6,38 @@
...
     "chart.js": "^2.9.3",
-    "copy-webpack-plugin": "^6.0.3",
-    "css-loader": "^3.6.0",
+    "copy-webpack-plugin": "^9.0.0",
+    "css-loader": "^5.2.6",
+    "css-minimizer-webpack-plugin": "^3.0.0",
     "datamaps": "^0.5.9",
-    "iframe-resizer": "^4.3.1",
-    "mini-css-extract-plugin": "^0.8.2",
-    "optimize-css-assets-webpack-plugin": "^5.0.3",
```
Comment (happysalada) [inline assets/package.json:25]: for webpack 5 this needs to be replaced with css-minimizer-webpack-plugin

File: assets/package.json [hunk 3]
```diff
@@ -6,38 +6,38 @@
...
+    "react-flatpickr": "3.10.5",
     "react-flip-move": "^3.0.4",
     "react-router-dom": "^5.2.0",
-    "react-transition-group": "^4.4.1",
-    "tailwindcss": "^2.0.1-compat",
-    "terser-webpack-plugin": "^4.2.3",
-    "url-search-params-polyfill": "^8.0.0",
-    "webpack": "4.39.2",
-    "webpack-cli": "^3.3.12"
+    "react-transition-group": "^4.4.2",
```
Comment (happysalada) [inline assets/package.json:35]: I don't know anything about this one, but it was a patch, so I thought it was fine to include.

File: assets/package.json [hunk 4]
```diff
@@ -6,38 +6,38 @@
...
-    "babel-loader": "^8.1.0",
+    "@babel/core": "^7.14.3",
+    "@babel/preset-env": "^7.14.4",
+    "@babel/preset-react": "^7.13.13",
+    "@tailwindcss/aspect-ratio": "^0.2.1",
+    "@tailwindcss/forms": "^0.3.2",
+    "@tailwindcss/typography": "^0.4.1",
+    "abortcontroller-polyfill": "^1.7.3",
+    "alpinejs": "^2.8.2",
+    "autoprefixer": "^10.2.6",
```
Comment (happysalada) [inline assets/package.json:17]: I'm not sure about this one.

**Comment by ukutaht** [comment]
Some conflicts have come up. Closing in favour of a rebased PR: https://github.com/plausible/analytics/pull/1112

**Pull request closed without merging**